### PR TITLE
Autoblock video with poster

### PR DESCRIPTION
**Pull request by sdmcraft**
Auto-blocks an image followed by url with text `//Video Link//` into an `embed` block with poster.
Example:
image

gets auto blocked as

image

Test URLs:
- Original: https://www.sunstar-foundation.org/<path>
- Before: https://main--sunstar-foundation--hlxsites.hlx.live/for-healthcare-info/education
- After: https://video-poster--sunstar-foundation--hlxsites.hlx.live/for-healthcare-info/education

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. cards | [doc-link](https://main--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=0)

- [ ] New variations introduced in this PR

Variation Name    | Documentation
------------- | -------------
 For e.g. cards (grid)  | [doc-link](https://main--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=2)

### CODE REVIEW

**Comment by RitwikSrivastava** [comment]
@sdmcraft On clicking on Play button , another play button needs to be pressed to actually play the video which doesnt seem very user friendly.

**Comment by jindaliiita** [comment]
> @sdmcraft On clicking on Play button , another play button needs to be pressed to actually play the video which doesnt seem very user friendly.

I was going to add the same comment.

**Comment by sdmcraft** [comment]
@jindaliiita , @RitwikSrivastava , the asset delivery recently added a video autoplay mechanism and I have incorporated it here and now the video auto-plays. So no double click needed. 
Also copying @ehrro

File: blocks/embed/embed.js [hunk 1]
```diff
@@ -142,6 +142,19 @@ const loadEmbed = (block, grandChilds, link, existingClassList) => {
     block.classList = `block embed embed-${config.match[0]}`;
   } else {
     block.innerHTML = getDefaultEmbed(url);
+    // Pass the video config to the iframe
+    const videoConfig = {
+      autoplay: 'muted',
```
Comment (harshchiki) [inline blocks/embed/embed.js:147]: Can we try with the value `any`?

with reference to https://videojs.com/guides/options/#autoplay.

**Review by jindaliiita — APPROVED** [review]

**Review by RitwikSrivastava — APPROVED** [review]

**Comment by RitwikSrivastava** [comment]
>the asset delivery recently added a video autoplay mechanism and I have incorporated it here and now the video auto-plays. So no double click needed.

Thanks @sdmcraft . This is also causing the embed block video to auto play on loading the page.

**Review by meejain — APPROVED** [review]

**Comment by sdmcraft** [comment]
> This is also causing the embed block video to auto play on loading the page.

Thanks. Yes I know. I need to place posters for those videos. The first video is the one which actually shows the proposed flow.

**Review by RitwikSrivastava — APPROVED** [review]

**Comment by jindaliiita** [comment]
@sdmcraft One problem with this change is that if there are multiple videos on a page, they will all start autoplaying.
Eg page - https://main--sunstar-foundation--hlxsites.hlx.live/en/awards/world-perio/videos

**Comment by jindaliiita** [comment]
Also one video is not rendering properly.
Screenshot 2024-02-06 at 4 09 03 PM

@sdmcraft

**Pull request merged**